src/ucs/datastruct/lru.h

```diff
@@ -0,0 +1,53 @@
+/**
+* Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2001-2023. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifndef __LRU_H__
+#define __LRU_H__
+
+#include <ucs/type/status.h>
+#include <stddef.h>
+
+typedef struct ucs_lru *ucs_lru_h;
+
+/**
+ * @brief Init a new LRU cache instance.
+ *
+ * @param [in]  capacity  Max number of elements to store.
+ * @param [out] lru_p     Pointer to the allocated lru struct.
+ *
+ * @return UCS_OK if successful, or an error code as defined by
+ * @ref ucs_status_t otherwise.
+ */
+ucs_status_t ucs_lru_create(size_t capacity, ucs_lru_h *lru_p);
+
+/**
+ * @brief Destroy LRU cache.
+ *
+ * @param [in] lru  Handle to the LRU cache.
+ */
+void ucs_lru_destroy(ucs_lru_h lru);
+
+/**
+ * @brief Insert or update an element in the cache.
+ *
+ * @param [in] lru  Handle to the LRU cache.
+ * @param [in] key  Element's key.
+ *
+ */
+void ucs_lru_put(ucs_lru_h lru, const void *key);
+
+/**
+ * @brief Get list of elements from the cache.
+ *
+ * @param [in]  lru   Handle to the LRU cache.
+ * @param [out] array Pointer to an allocated array to store elements.

```

Comment:
Array does not "contain" bytes, but is rather "of size X bytes" or "X bytes big".
Anyway, I somewhat dislike this interface (user array size requirement). Can suddenly break & cause mess. Can think of something else? Maybe the user will pass the array size?